### PR TITLE
Test updates for https://github.com/dotnet/coreclr/pull/16447

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Updates and disables tests, tracked by #27269.  For Path change in CoreCLR: https://github.com/dotnet/coreclr/pull/16447

### CODE REVIEW

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs [hunk 1]
```diff
@@ -36,11 +36,20 @@ public static void ChangeExtension(string path, string newExtension, string expe
...
         [Fact]
-        public static void GetDirectoryName_EmptyThrows()
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)]
+        public static void GetDirectoryName_EmptyThrows_Desktop()
         {
             AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetDirectoryName(string.Empty));
         }
 
+        [Fact]
+        [ActiveIssue(27269)]
```
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: #27269 seems to be they issue for this PR. Wrong number?

Comment (JeremyKuhne) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: That should be right. This is disabling the test. I'll change the issue to re-enable after corelib is consumed.

**Comment by Anipik** [comment]
We should add tests for these cases https://github.com/dotnet/coreclr/pull/16447/files#diff-9e997eca42c8d36538cb7a1cacc62e91R182

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs [hunk 2]
```diff
@@ -60,19 +69,10 @@ public static void GetDirectoryName_SpaceOrControlCharsThrowOnWindows(string pat
...
-            if (PlatformDetection.IsWindows)
-            {
-                AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetDirectoryName(path));
-            }
-            else
-            {
-                // This is a valid path on Unix
-                action();
-            }
+            Assert.Null(Path.GetDirectoryName(" "));
```
Comment (Anipik) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: We dont have a test for this ```"\r\n"``` on .net core

Comment (JeremyKuhne) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: Thanks for pointing it out. I'll be making several passes through the tests as we settle in the changes in CoreLib. Please feel free to create new tests where you see we're lacking as well. :)

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Linux x64 Release Build

```
Message :
Expecting Child Task final Status to be Canceled, while getting RanToCompletion
Expected: True
Actual:   False
Stack Trace :
   at System.Threading.Tasks.Tests.Status.TaskStatusTest.RealRun() in /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Threading.Tasks/tests/Task/TaskStatusTest.cs:line 257
```

**Pull request merged**